### PR TITLE
add advance conf for otel-collector 

### LINKED ISSUE #174: Add some "demos" for collector components
# Feature Request

The current [collector configuration](https://github.com/open-telemetry/opentelemetry-demo-webstore/blob/main/src/otelcollector/otelcol-config.yml) coming with the demo-webstore is mostly for receiving traces/metrics and exporting them once again. 

I think there's some opportunity to easily add some demos of collector features/components, e.g.

* use some processors like attributesprocessor/transformprocessor to add/delete/... attributes, maybe some of the services are exposing sensitive data (like username) and the processor is dropping that data.
* compute/transform some metrics
* use a sampling processor
* receive metrics from redis?
* have a "legacy" service that exports traces/metrics in a different format 
* ...

There could be a second otelcol-advanced-config.yml or those settings could be disabled by default.

### LINKED ISSUE #192: change docker-compse image pull way
# Feature Request

the docker-compse current use `build` command to create the image, it often occur unexpected problem when someone want to  take a look the demo, like #178 #170 #126, and I notice #184 had merged, can we use the prebuild image instead of build localy? by the way, there is also don't have a clearly way to kown how to find the demo image  in docker hub.

### PR DESCRIPTION
**Pull request by Frapschen**
Fixes #174 

I think the spanmetrics processor is one of a good demo for advance ability of otel-collector.

## Changes

* add a spanmetrics processor in otel-collector
* use otel/opentelemetry-collector-contrib:0.52.0 image to support advance ability
* use released images in docker-compose

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #

### CODE REVIEW

**Review by wph95 — CHANGES_REQUESTED** [review]
I saw you change some file
- add jaeger-ui.json
- change otel-collector config
- change docker compose image name logic

One pr per feature, which will make the review more efficient.

**Comment by Frapschen** [comment]
> I saw you change some file
> 
> * add jaeger-ui.json
> * change otel-collector config
> * change docker compose image name logic
> 
> One pr per feature, which will make the review more efficient.

sorry, I have recovered the on relate change

File: src/otelcollector/otelcol-config.yml [hunk 1]
```diff
@@ -16,14 +20,19 @@ exporters:
...
   batch:
+  spanmetrics:
+    metrics_exporter: prometheus
 
 service:
   pipelines:
     traces:
       receivers: [otlp]
-      processors: [batch]
+      processors: [spanmetrics,batch]
```
Comment (cijothomas) [inline src/otelcollector/otelcol-config.yml:30]:
```suggestion
      processors: [spanmetrics, batch]
```

File: src/otelcollector/otelcol-config.yml [hunk 2]
```diff
@@ -16,14 +20,19 @@ exporters:
...
       exporters: [logging, jaeger]
     metrics:
       receivers: [otlp]
       processors: [batch]
       exporters: [prometheus, logging]
+    metrics/spanmetrics:
+      receivers: [ otlp/spanmetrics ]
+      exporters: [ prometheus ]
```
Comment (puckpuck) [inline src/otelcollector/otelcol-config.yml:38]: Can we add the logging exporter to this as well?

**Review by puckpuck — CHANGES_REQUESTED** [review]
Some small nitpick changes on the Otel Config side, but also would want to see all the Jaeger changes removed from this PR.

File: src/otelcollector/otelcol-config.yml (hunk 2 above)
Comment (Frapschen) [reply src/otelcollector/otelcol-config.yml]: done

**Comment by Frapschen** [comment]
> Some small nitpick changes on the Otel Config side, but also would want to see all the Jaeger changes removed from this PR.

Jaeger have a monitor page to show the span metrics, I think we can ues it.

**Comment by cartersocha** [comment]
> > Some small nitpick changes on the Otel Config side, but also would want to see all the Jaeger changes removed from this PR.
> 
> Jaeger have a monitor page to show the span metrics, I think we can ues it.

Our preference / design is native OTel metrics only with maybe Prometheus if needed consumed in grafana. We’re not planning on using jaeger for metrics in the main demo & by default we’d like it to be off. Users are welcome to enable it on their own if they choose

File: docker-compose.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ services:
 
   # Collector
   otelcol:
-    image: otel/opentelemetry-collector:0.56.0
+    image: otel/opentelemetry-collector-contrib:0.52.0
```
Comment (julianocosta89) [inline docker-compose.yml:29]: Can we have collector-contrib 0.56.0?
```suggestion
    image: otel/opentelemetry-collector-contrib:0.56.0
```

**Review by julianocosta89 — APPROVED** [review]
Just a version bump, but looks good to me!

**Review by puckpuck — APPROVED** [review]

**Review by mic-max — APPROVED** [review]

**Comment by cartersocha** [comment]
@Frapschen please put a changelog entry for this & we'll merge!

File: CHANGELOG.md [hunk 4]
```diff
@@ -53,3 +53,5 @@ significant modifications will be credited to OpenTelemetry Authors.
 ([#253](https://github.com/open-telemetry/opentelemetry-demo/pull/253))
 * Added explicit support for Kubernetes.
 ([#255](https://github.com/open-telemetry/opentelemetry-demo/pull/255))
+* Added spanmetrics processor to otelcol
+([#212]((https://github.com/open-telemetry/opentelemetry-demo/pull/212)))
```
Comment (wph95) [inline CHANGELOG.md:57]:
```suggestion
([#212](https://github.com/open-telemetry/opentelemetry-demo/pull/212))
```

Comment (wph95) [reply CHANGELOG.md:57]: typo need fix

**Review by wph95 — APPROVED** [review]
LGTM

**Pull request merged**